SimCalorimetry/HGCalSimProducers/src/HGCDigitizer.cc

```diff
@@ -142,7 +162,30 @@ void HGCDigitizer::finalizeEvent(edm::Event& e, edm::EventSetup const& es, CLHEP
   const CaloSubdetectorGeometry* theGeom = ( nullptr == gHGCal_ ? 
 					     static_cast<const CaloSubdetectorGeometry*>(gHcal_) : 
 					     static_cast<const CaloSubdetectorGeometry*>(gHGCal_)  );
-
+  
+  ++nEvents_;
+  unsigned idx = std::numeric_limits<unsigned>::max();
+  switch(mySubDet_) {
+  case ForwardSubdetector::HGCEE:
+    idx = 0;
+    break;
+  case ForwardSubdetector::HGCHEF:
+    idx = 1;
+    break;
+  case ForwardSubdetector::HGCHEB:
+    idx = 2;
+    break;
+  default:
+    break;
+  }
+  // release memory for unfilled parts of hash table
+  if( validIds_.size()*averageOccupancies_[idx] > simHitAccumulator_->size() ) {
+    simHitAccumulator_->reserve(simHitAccumulator_->size());
```

Comment:
Sorry to raise an old PR, but what is the aim of this line? I'd expect this to be a no-op.

(and what would be the better way to ask random questions on random pieces of code?)